Networking/WebSockets/WebSocketImpl.cc

```diff
@@ -453,6 +461,15 @@ namespace litecore { namespace websocket {
 
     // Called when the underlying socket closes.
     void WebSocketImpl::onClose(CloseStatus status) {
+        CHECK_DELEGATE
+        // CBL-2751
+        // The delegate is assigned when WebSocket::connect(Delegate*) is called,
+        // which is an act of client.
```

Comment:
That's not true. Both client and server side call connect(). And any active WebSocket has to have a delegate, because incoming messages are sent to it.

You can find the call hierarchy with Xcode, for example:
* WebSocket::connect
* BLIPIO::_start
* BLIPIO::start
* blip::Connection::start
* Replicator::_start
* Replicator::start
* C4ReplicatorImpl::_start
* C4ReplicatorImpl::start
* RESTSyncListener::handleSync

Reply:
Thanks, they are so helpful. Your's saying, if not active, the argument of WebSocket::connect(Delegate*) is null. Am I right?

Reply:
But there is an assertion asserts none-null.